src/bin/pgcopydb/dump_restore.c

```diff
@@ -622,31 +624,64 @@ copydb_write_restore_list(CopyDataSpec *specs, PostgresDumpSection section)
 	 *   2. edit post.list to comment out lines and save as filtered.list
 	 *   3. pg_restore --section post-data --use-list filtered.list schema.dump
 	 */
-	ArchiveContentArray contents = { 0 };
-
 	if (!pg_restore_list(&(specs->pgPaths),
 						 dumpFilename,
-						 listOutFilename,
-						 &contents))
+						 listOutFilename))
 	{
 		/* errors have already been logged */
 		return false;
 	}
 
 	/* edit our post.list file now */
-	PQExpBuffer listContents = createPQExpBuffer();
-
-	if (listContents == NULL)
+	PQExpBuffer line = createPQExpBuffer();
```

Comment:
A single pg_restore archive TOC item has a maximum known length. Consider switching to local static allocation instead; or dynamic allocation of a known size.